File: problems/1231-replace-elements-with-greatest-element-on-right-side/solution.cpp
class Solution {
public:
    vector<int> replaceElements(vector<int>& arr) {
        int last = arr.size() - 1;
        int max = arr[last];
        arr[last] = -1;
        last--;
        
        for (int i = last; i >= 0; i--) {
            int temp = arr[i];
            arr[i] = max;
            max = max > temp ? max : temp;
        }
        return arr;
    }
};

File: problems/1231-replace-elements-with-greatest-element-on-right-side/solution.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        rtlMax = []
        mx = 0
        for i in range(len(arr) - 1, -1, -1):
            if i == len(arr) - 1:
                rtlMax.append(-1)
                continue

            mx = max(mx, arr[i + 1])
            rtlMax.append(mx)
        rtlMax.reverse()

        return rtlMax

            
            
